test/Table_Tests/src/Expression_Spec.enso

```diff
@@ -0,0 +1,156 @@
+from Standard.Base import all
+from Standard.Base.Error.Problem_Behavior import Report_Error
+
+from Standard.Table import Table, Column, Sort_Column, Column_Selector, Sort_Column_Selector, Aggregate_Column
+
+import Standard.Visualization
+
+from Standard.Test import Test, Test_Suite, Problems
+
+import project.Common_Table_Spec
+from project.Util import all
+
+spec detailed=False =
+    # Create Test Table
+    column_a = ["A", [1, 2, 3, 4, 5]]
+    column_b = ["B", [1.0, 1.5, 2.5, 4, 6]]
+    column_odd = ["Bad] Name", [True, False, True, False, True]]
+    test_table = Table.new [column_a, column_b, column_odd]
+
+    epsilon=0.0000000001
+
+    tester expression value =
+        new_table = test_table.set "NEW_COL" expression
+        new_table.column_count . should_equal (test_table.column_count + 1)
+        new_column = new_table.at -1
+        new_column.name . should_equal "NEW_COL"
+        expected = case value of
+            _ : Vector.Vector -> value
+            _ -> Vector.new test_table.row_count _->value
+
+        values = new_column.to_vector
+        values . each_with_index i->v->
+            e = expected.at i
+            match = case e of
+                _ : Number -> e.equals v epsilon
+                _ -> e == v
+            if match then Nothing else
+                values.should_equal expected
+
+    specify_test specify action =
+        case detailed of
+            True ->
+                specify_tester expression value =
+                    Test.specify (specify + ": " + expression) <|
+                        tester expression value
+                action specify_tester
+            False ->
+                Test.specify specify (action tester)
+
+    Test.group "Expression Integer literals" <|
+        specify_test "should be able to add an integer column" expression_test->
+            expression_test "1" 1
+            expression_test "-3" -3
+            expression_test "1_000" 1000
+
+    Test.group "Expression Decimal literals" <|
+        specify_test "should be able to add an decimal column" expression_test->
+            expression_test "1.23" 1.23
+            expression_test "-3.1415" -3.1415
+            expression_test "1_000.456" 1000.456
+
+    Test.group "Expression Boolean literals" <|
+        specify_test "should be able to add a boolean column" expression_test->
+            expression_test "True" True
+            expression_test "true" True
+            expression_test "TRUE" True
+            expression_test "False" False
+            expression_test "false" False
+            expression_test "FALSE" False
+
+    Test.group "Expression Text literals" <|
+        specify_test "should be able to add a text column" expression_test->
+            expression_test "'Hello World'" 'Hello World'
+            expression_test "'Hello \'World\''" "Hello 'World'"
+            expression_test '"Hello World"' 'Hello World'
+            expression_test '"Hello ""World"""' 'Hello "World"'
```

Comment:
I would add a bit more tests testing the Python escapes here. Also would be good to test the Excel string that it just does not do anything unexpected with `\`.